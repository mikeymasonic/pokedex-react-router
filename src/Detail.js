// import React, { Component } from 'react';
// import { getCharacter } from './api.js'
// import CharacterItem from './CharacterItem.js';

// export default class Detail extends Component {
//     state = { character: {} }

//     async componentDidMount() {
//         const data = await getCharacter(this.props.match.params.charId);
        
//         if (data.body.results) {


//         this.setState({ character: data.body.results[0] })
            
//         }
//     }

//     render() {
//         const { character } = this.state;

//         return (
//             <CharacterItem character={ character } />
//       );
//     }
// }

import React, { Component } from 'react'
import { getPokemonById } from './api.js';
import PokeItem from './PokeItem.js';

export default class Detail extends Component {

    state = { pokemonChosen: {} }

    async componentDidMount() {
        const pokeOnline = await getPokemonById(this.props.match.params.pokeid);
        this.setState({ pokemonChosen: pokeOnline.body });
    }

    render() {
        return (
            <div>
                <ul className="pokes">
                    <PokeItem pokemon={this.state.pokemonChosen} />
                </ul>
            </div>
        )
    }
}
