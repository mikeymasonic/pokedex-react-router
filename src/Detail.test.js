import React from 'react';
import Detail from './Detail';
import renderer from "react-test-renderer";
test('renders Detail.js correctly', () => {
  const tree = renderer
  .create(<Detail pokeDex={[]}/>)
  .toJSON();
  expect(tree).toMatchSnapshot();
});