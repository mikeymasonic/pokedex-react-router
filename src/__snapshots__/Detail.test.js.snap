// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Detail.js correctly 1`] = `
<div>
  <ul
    className="pokes"
  >
    <li>
      <div
        className="info-container"
      >
        <h2 />
        <p
          className="poke-hp"
        >
          <b>
            HP:
          </b>
           
           / 
          <b>
            XP:
          </b>
           
        </p>
      </div>
      <div
        className="image-container"
      >
        <img />
      </div>
      <p
        className="attributes-class"
      >
        <span
          id="type"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        >
          <b>
            Type: 
          </b>
           
        </span>
        <br />
        <b>
          Attack:
        </b>
         
         /
        <b>
           Defense:
        </b>
         
        <br />
        <b>
           Special Attack:
        </b>
         
         /
        <b>
           Special Defense:
        </b>
         
        <br />
        <b>
           Speed:
        </b>
         
        <br />
        <b>
           Height:
        </b>
         
        " /
        <b>
           Weight:
        </b>
         
         oz
      </p>
    </li>
  </ul>
</div>
`;
